### PR TITLE
Some miscellaneous fixes from scan-build test

### PR DESCRIPTION
**Pull request by lealem47**
# Description

Fixes for nighly scan-build test

Fixes zd#

# Testing

How did you test?

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfcrypt/src/kdf.c [hunk 1]
```diff
@@ -313,8 +313,12 @@ int wc_PRF_TLS(byte* digest, word32 digLen, const byte* secret, word32 secLen,
         byte labelSeed[MAX_PRF_LABSEED];
     #endif
 
-        if (labLen + seedLen > MAX_PRF_LABSEED)
+        if (labLen + seedLen > MAX_PRF_LABSEED) {
+        #if defined(WOLFSSL_ASYNC_CRYPT) && !defined(WC_ASYNC_NO_HASH)
```
Comment (dgarske) [inline wolfcrypt/src/kdf.c]: I refactored this code and also resolve another place that could leak in KDF.

**Review by haydenroche5 — APPROVED** [review]

**Comment by lealem47** [comment]
Hold off on merge. A couple of minor changes are still outstanding.

**Comment by haydenroche5** [comment]
> Hold off on merge. A couple of minor changes are still outstanding.

Cool. Just assign to me when ready.

File: src/internal.c [hunk 2]
```diff
@@ -29371,7 +29374,8 @@ int SendClientKeyExchange(WOLFSSL* ssl)
                 case ecc_diffie_hellman_kea:
                 {
                     /* place size of public key in buffer */
-                    *args->encSecret = (byte)args->encSz;
+                    if (args->encSecret != NULL)
```
Comment (haydenroche5) [inline src/internal.c:29377]: Should we error out if it is somehow NULL?

File: src/tls.c [hunk 3]
```diff
@@ -11959,7 +11959,9 @@ int TLSX_PopulateExtensions(WOLFSSL* ssl, byte isServer)
             ret = TLSX_PopulateSupportedGroups(ssl, &ssl->extensions);
             if (ret != WOLFSSL_SUCCESS)
                 return ret;
+        #ifdef NO_PSK
```
Comment (haydenroche5) [inline src/tls.c:11962]: Curious what this is addressing. Could be worth adding a comment explaining.

**Review by haydenroche5 — CHANGES_REQUESTED** [review]

File: src/internal.c (hunk 2 above)
Comment (lealem47) [reply src/internal.c:29377]: Good idea. Preferred ret value for this case?

File: src/tls.c (hunk 3 above)
Comment (lealem47) [reply src/tls.c:11962]: That value just ends up never being read in the `!NO_PSK` case. I'll add a comment.

File: src/internal.c (hunk 2 above)
Comment (haydenroche5) [reply src/internal.c:29377]: For these sorts of things that should "never happen," BAD_STATE_E is one option.

Comment (lealem47) [reply src/internal.c:29377]: Done

File: src/tls.c (hunk 3 above)
Comment (lealem47) [reply src/tls.c]: Done

**Review by haydenroche5 — APPROVED** [review]

**Pull request merged**